### DISCUSSION TITLE
Boost text search with silver searcher (ag)

### DISCUSSION THREAD

**Discussion by Sylvain303** [zim-desktop-wiki/zim-desktop-wiki#1940] [Ideas]
Hello,

I'm using Zim since a lot of years as my personal knowledge base. It grew up a lot:

```
$ cd ~/Notebooks/Notes
$ wc  $(find . -type f -name \*.txt) | tail -1
 139139  722712 5700930 total
$ find . -type d | wc -l
487
$ find . -type f | wc -l
2749
```

I often use Zim's text search, that became a bit slow due to the vast content collected.

On the counter part I discovered `ag` [the Silver searcher](https://github.com/ggreer/the_silver_searcher), that's is just the faster text search I ever met. (instantaneous on my Zim note book)

I would be great to have a configuration to swap to `ag` search for text search, if the software is installed.

**Comment by introt** [comment]
The topic has previously been discussed over at #64, with the conclusion that third-party search "engines" shall be supported via plugins after the necessary APIs have been made available.

If you're feeling adventurous and know your way around Python, you can try swapping `grep` for `ag` in [this patch](https://github.com/introt/zim-desktop-wiki/commit/28ed20e65c8157baea607587d1f027d23dc5d924).

As someone with a wiki in the same ballpark as yours, I would also recommend trying out the instant search plugin (link & some background [here](https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/64#issuecomment-1002521828)). While Zim's current search implementation is on the slow side, the plugin proves that there are a number of ways to improve performance without calling external tools.

**Reply by Sylvain303** [reply]
Ah, ok, good to know. I will try to hack on that. Thanks.

I will comeback when I will have some news about that.
